examples/finetune/finetune.cpp

```diff
@@ -1768,11 +1769,11 @@ int main(int argc, char ** argv) {
     for (unsigned order = 0; order < (unsigned) GGML_CGRAPH_EVAL_ORDER_COUNT; ++order) {
         ctx_compute = ggml_init(ctx_compute_params);
         alloc = ggml_allocr_new_measure(tensor_alignment);
-        gf = ggml_new_graph(ctx_compute);
+        gf = ggml_new_graph_custom(ctx_compute, LLAMA_TRAIN_MAX_NODES, true);
         gf->order = (enum ggml_cgraph_eval_order) order;
-        gb = ggml_new_graph(ctx_compute);
+        gb = ggml_new_graph_custom(ctx_compute, LLAMA_TRAIN_MAX_NODES, false);
         gb_tmp = params.common.use_checkpointing
-            ? ggml_new_graph(ctx_compute)
+            ? ggml_new_graph_custom(ctx_compute, LLAMA_TRAIN_MAX_NODES, false)
```

Comment:
@slaren Does this look OK?

Reply:
I think so, I don't know if `gb` here needs grads or not.

Reply:
`gb` needs grads, because `gb` also contains the `gf` nodes, which have grads.

Changing the `bool grads` from `false` to `true` resolves a triggered assert in ggml.c `ggml_graph_cpy`.
```cpp
GGML_ASSERT(dst->grads != NULL);
```

With this change finetune runs, I will report back if the results are good as well.

Reply:
Should `ggml_graph_cpy` be changed to allow skipping the grads if the src has then but not the dst?

Reply:
There was an additional - unrelated to this PR - regression in finetune and train-text-from-scratch due to new yarn rope implementation.

Changing `bool grads` argument to true and applying #3974 to fix the backward process of rope, the output of finetune is correct.

Reply:
Just to note, lines 1805 and 1807 needs that change as well, I missed them at first attempt to copy this fix.
Also, mentioned regression and #3974 fix seems to be critical, because otherwise finetune produces LORA's without any progress from one checkpoint to another.